static/js/actions/ui.js

```diff
@@ -22,9 +23,12 @@ export const setDialogVisibility = (bool: boolean): Action => (
 
 // work history actions
 export const SET_WORK_HISTORY_EDIT = 'SET_WORK_HISTORY_EDIT';
-export const setWorkHistoryEdit = (bool: boolean): Action => (
-  { type: SET_WORK_HISTORY_EDIT, payload: bool }
-);
+export const setWorkHistoryEdit: Function = (bool: boolean): (d: Dispatch) => Promise => {
```

Comment:
2 things:

1. can we change the name of `bool` to `flag` or something? it's a nitpick for such a small function, but i think it's a little more descriptive and 'flag' isn't a type declaration in other languages
1. this declaration syntax is a little mind-bending. can you explain in english just for my benefit? some sources of confusion: `d` in `(d: Dispatch)` doesn't appear to be used, `(d: Dispatch)` seems like it might be a return type, but then `Promise` looks like the actual return type.